src/hssm/param/regression_param.py

```diff
@@ -0,0 +1,402 @@
+    fill_defaults
+        Fill in the default values for the parameter.
+    validate
+        Validate the parameter.
+    process_prior
+        Process the prior specification.
+    set_loglogit_link
+        Override the default link function.
+    make_safe_priors
+        Override the default priors.
+    reformat_formula
+        Reformat the formula.
+    """
+
+    terms: list[str]
+
+    def __init__(
+        self,
+        name: str,
+        formula: str | None = None,
+        prior: dict[str, Any] | None = None,
+        link: str | Link | None = None,
+        bounds: tuple[float, float] | None = None,
+        user_param: UserParam | None = None,
+    ) -> None:
+        super().__init__(
+            name, prior, formula, link, bounds=bounds, user_param=user_param
+        )
+        self.terms = []
+
+    @classmethod
+    def from_defaults(
+        cls,
+        name: str,
+        formula: str,
+        bounds: tuple[float, float] | None = None,
+        link_settings: Literal["log_logit"] | None = None,
+    ) -> "RegressionParam":
+        """Create a RegressionParam object with default values."""
+        param = cls(
+            name,
+            formula=formula,
+            prior=None,
+            link=None,
+            bounds=bounds,
+            user_param=None,
+        )
+        if link_settings == "log_logit":
+            param.set_loglogit_link()
+        return param
+
+    def fill_defaults(
+        self,
+        prior: dict[str, Any] | None = None,
+        bounds: tuple[float, float] | None = None,
+        **kwargs,
+    ) -> None:
+        """Fill in the default values for the parameter.
+
+        Parameters
+        ----------
+        formula
+            The formula for the parameter.
+        bounds
+            The bounds for the parameter.
+        prior_settings
+            The prior settings for the parameter.
+        link_settings
+            The link settings for the parameter.
+        """
+        if prior is not None:
+            raise ValueError(
+                f"{self.name} is a regression parameter. "
+                + "It should not have a default prior."
+            )
+        super().fill_defaults(prior=None, bounds=bounds)
+        if self.formula is None:
+            if "formula" not in kwargs:
+                raise ValueError(f"Formula not specified for parameter {self.name}.")
+            self.formula = kwargs["formula"]
+        if kwargs.get("link_settings") == "log_logit":
+            self.set_loglogit_link()
+
+    def validate(self) -> None:
+        """Validate the parameter."""
+        if self.formula is None:
+            raise ValueError(f"Formula not specified for parameter {self.name}.")
+        self.reformat_formula()
+        if isinstance(self.prior, bmb.Prior):
+            raise ValueError(
+                "Please specify priors for each individual parameter in the "
+                + f"regression for {self.name}."
```

Comment:
I think you can drop the `+` here.
```suggestion
                "Please specify priors for each individual parameter in the "
                f"regression for {self.name}."
```